src/ucs/sys/topo.h

```diff
@@ -0,0 +1,80 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TOPO_H
+#define UCS_TOPO_H
+
+#include <ucs/type/status.h>
+#include <limits.h>
+#include <stdio.h>
+
+BEGIN_C_DECLS
+
+/** @file topo.h */
+
+typedef struct ucs_sys_bus_id {
+    unsigned int domain;   /* range: 0 to ffff */
+    unsigned int bus;      /* range: 0 to ff */
+    unsigned int slot;     /* range: 0 to 1f */
+    unsigned int function; /* range: 0 to 7 */
+} ucs_sys_bus_id_t;
+
+
+/**
+ * @ingroup UCS_RESOURCE
+ * System Device abstraction
+ */
+typedef struct ucs_sys_device {
+    unsigned int id;               /**< Index of the unit */
+    int numa_node;                 /**< NUMA node assoicated with the device*/
+    const ucs_sys_bus_id_t bus_id; /**< bus ID of of the device if applicable.
+                                   eg: 0000:06:00.0 {domain:bus:slot.function}*/
+} ucs_sys_device_t;
+
+
+/*
+ * Capture the estimated latency, bandwidth between two system devices
+ * referred by ucs_sys_device_t handle
+ */
+typedef struct ucs_sys_dev_distance {
+    double latency;   /**< in seconds */
+    double bandwidth; /**< in bytes/second */
+} ucs_sys_dev_distance_t;
+
+
+/**
+ * Find system device by pci bus id
+ *
+ * @param [in]     bus_id  bus id of the device of interest
+ * @param [in/out] sys_dev pointer to ucs_sys_device_t
+ *                         populated with device associated with the bus_id
+ *
+ * @return UCS_OK or error in case device cannot be found
+ */
+ucs_status_t ucs_topo_find_device_by_bus_id(const ucs_sys_bus_id_t *bus_id, const ucs_sys_device_t *sys_dev);
```

Comment:
line too long
sys_dev is returned, it should be `const ucs_sys_device_t **device`